### ISSUE TITLE
Improve Prop types for Button Component

### ISSUE THREAD

**Issue by ievavizg** [nordcloud/GNUI#422]
Prop types need to be improved, so Typescript takes props from element passed as polymorphic component ("as").
This could be tricky considering `linkTo` prop.
Useful resource: https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase#polymorphic-components-eg-with-as-props

`download` attribute should be available when we pass anchor as a polymorphic element.
```tsx
<Button as="a" download />
```

It shouldn't be available when the default element ("button") is used, e.g.
```tsx
// This throws error
<Button download />
```

**Comment by nowyDEV** [comment]
Prop types need to be improved, so Typescript takes props from element passed as polymorphic component ("as").
This could be tricky considering `linkTo` prop.
Useful resource: https://react-typescript-cheatsheet.netlify.app/docs/advanced/patterns_by_usecase#polymorphic-components-eg-with-as-props

**Comment by nowyDEV** [comment]
Resolved in #641
